src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java

```diff
@@ -420,4 +451,88 @@ public static boolean isCorrectJavadocPosition(DetailAST blockComment) {
                 || BlockCommentPosition.isOnEnumConstant(blockComment)
                 || BlockCommentPosition.isOnAnnotationDef(blockComment);
     }
+
+    /**
+     * Checks if the supplied {@code DetailNode} argument is not a tight HTML tag.
+     * @param node the node to check
+     * @return true if the javadoc node is not a tight HTML tag; false otherwise
+     */
+    public static boolean isNonTightHtmlTag(DetailNode node) {
+        final boolean isLooseHtmlTag;
+        isLooseHtmlTag = node.getText().endsWith("TAG_START")
+                && node.getParent().getType() != getHtmlTagParentTypeWhenClosed(node);
```

Comment:
@romani Please have a look at [3311#issuecomment](https://github.com/checkstyle/checkstyle/issues/3311#issuecomment-312201575). Also adding unnecessary code to java parser might unpredictably slow it down further, why burden it ? 

As @baratali has pointed in the issue description, to make the grammar set flag for non tight tags we will need to add code to all type of `...TagStart` (`...TagOpen`) rules. This option is much more better and efficient, I feel.